cmd/mdatagen/metricdata.go

```diff
@@ -61,8 +61,7 @@ type Mono struct {
 // NumberDataPoints defines the metric number type.
 type NumberDataPoints struct {
 	// Type is type of the metric number, options are "double", "int".
-	// TODO: Add validation once the metric number type added to all metadata files.
-	NumberType string `mapstructure:"number_type"`
+	NumberType string `mapstructure:"number_type" validate:"oneof=double int"`
```

Comment:
Does "validate" tag checks for the enum values? In the collector we don't use this tag, may be good to look into that if it is that nice.

Alternative you can have a `type NumberType string` and implement https://pkg.go.dev/encoding#TextUnmarshaler where you accept only the values you want. See https://github.com/open-telemetry/opentelemetry-collector/blob/bd13c61ee2fcfc335e81218e8c6554ba893d3532/config/identifiable.go#L73

Reply:
Makes sense. Updated

Reply:
`validate` tag is being used for validation only, not for unmarshalling. In order to use enum we need to provide a custom unmarshaler. I did it with `TextUnmarshaler` similar to what you suggested. PTAL